pandas/tests/series/methods/test_replace.py

```diff
@@ -402,3 +402,20 @@ def test_replace_only_one_dictlike_arg(self):
         msg = "Series.replace cannot use dict-value and non-None to_replace"
         with pytest.raises(ValueError, match=msg):
             ser.replace(to_replace, value)
+
+    @pytest.mark.parametrize(
+        "series, to_replace, expected",
+        [
+            (
+                pd.Series(["one", "two"], dtype="string"),
+                {"one": "1", "two": "2"},
```

Comment:
can you add a bunch more tests which can hit the other EA types, e.g. Intervval, Period, Datetime w/tz

Reply:
I added test cases.